src/cascadia/TerminalSettingsModel/Command.cpp

```diff
@@ -383,6 +384,65 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         return warnings;
     }
 
+    // Function Description:
+    // - Serialize the Command into an array of json actions
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - an array of serialized actions
+    Json::Value Command::ToJson() const
+    {
+        Json::Value cmdList{ Json::ValueType::arrayValue };
+
+        if (_nestedCommand)
+        {
+            // handle nested command
+            // For nested commands, we can trust _originalJson to be correct.
+            // In fact, we _need_ to use it here because we don't actually deserialize `iterateOn`
+            //   until we expand the command.
+            cmdList.append(_originalJson);
+        }
+        else if (_keyMappings.empty())
+        {
+            // only write out one command
+            Json::Value cmdJson{ Json::ValueType::objectValue };
+            JsonUtils::SetValueForKey(cmdJson, IconKey, _iconPath);
+            JsonUtils::SetValueForKey(cmdJson, NameKey, _name);
+
+            if (_ActionAndArgs)
+            {
+                cmdJson[JsonKey(ActionKey)] = ActionAndArgs::ToJson(_ActionAndArgs);
+            }
+
+            cmdList.append(cmdJson);
+        }
+        else
+        {
+            // we'll write out one command per key mapping
+            for (auto keys{ _keyMappings.begin() }; keys != _keyMappings.end(); ++keys)

```

Comment:
```suggestion
            for (const auto keys{ _keyMappings.begin() }; keys != _keyMappings.end(); ++keys)
```

Reply:
I'll do you one better:
```cpp
for (const auto& keys : _keyMappings)
```

Reply:
BTW, `const` didn't work on this because of the `++keys` at the end